python/cuml/common/base.pyx

```diff
@@ -282,6 +282,51 @@ class Base:
             else:
                 raise AttributeError
 
+    def _set_base_attributes(self,
+                             X,
+                             output_type=False,
+                             y=None,
+                             n_features=None):
+        """
+        Method to set the base class attributes - output type,
+        target dtype and n_features. It combines the three different
+        function calls. It's called in fit function from estimators.
+
+        Parameters
+        --------
+        X : cuDF DataFrame
+            Specifies the input data training
+        output_type : Boolean (default = False)
+            Specifies if we should set output type on X
```

Comment:
Following from the above, this could be a single parameter `output_type` whose default is `None`, right?

Reply:
No, it's actually the output type of X. Some models only use self._set_output_type(X) instead of on y which makes.

Reply:
then the name/behavior is confusing. From reading the API what attribute is `X` setting? I think we should make all the parameter names reflect the attributes they set, right now it is very confusing to have `X` and then `output_type` which is a boolean